src/test/java/dev/thomazz/pledge/ClientPingerTests.java

```diff
@@ -110,17 +110,17 @@ public void testFrameClientPinger() {
         for (int i = 0; i > -400; i--) {
             Optional<Frame> frame;
             if (toggle) {
-                frame = frameData.matchStart(i);
+                frame = frameData.matchStart(i, (f) -> {});
```

Comment:
because the consumer is a handler for handling multiple frames( frame skipping). It should not happen in unit test

Reply:
Can you explain why you would want to process multiple frames at the same time? I'm having difficulty understanding what this would be useful for 🤔

Reply:
If the frame is skipped, then the skipped frame also needs to be processed, so a consumer is needed.

Reply:
I don't agree with processing frames that have not been received from the client. It will break existing implementations that assume this behavior to be impossible.

Reply:
I disagree with this change, the order of frame should be kept and we shouldn't process anything until we received the corresponding ID of the oldest pending frame.